File: src/unified_data.py
import glob
from pathlib import Path

from duckdb import DuckDBPyRelation
from core import *

p_country_of_publication = cast(nw.LazyFrame[DuckDBPyRelation], None)
p_title = cast(nw.LazyFrame[DuckDBPyRelation], None)
p_year_of_publication = cast(nw.LazyFrame[DuckDBPyRelation], None)

for row in glob.glob(str(here("data/unified/*.parquet"))):
    name = Path(row).stem
    globals()[name] = read_parquet(name, here(f"data/unified/{name}.parquet"))
    print(f"{name} = cast(nw.LazyFrame[DuckDBPyRelation], None)")


__all__ = ["p_country_of_publication", "p_title", "p_year_of_publication"]

File: src/unified_data.R
box::use(src/core[con])

Sys.glob(here::here("data/unified/*.parquet")) |>
  purrr::walk(\(path) {
    t <- box::topenv()
    name = tools::file_path_sans_ext(basename(path))
    t[[name]] <- dplyr::tbl(con, glue::glue("read_parquet('{here::here('data/unified')}/{name}*.parquet')"))
  })
